### DISCUSSION TITLE
Proposal: publish beta/rc versions to GitHub Package Registry

### DISCUSSION THREAD

**Discussion by o-az** [family/connectkit#335] [Ideas]
I'm following your https://github.com/family/connectkit/tree/wagmi-2.x progress and proposing that you start publishing beta releases on push, it'll help others test out the beta release and provide feedback.

Here's my proposal:

- Use [GitHub Package Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry) for non-main releases. Continue using npm for main releases - so your npm publishing flow remains unchanged.
- GitHub Packages work the same as npm, just a different registry. Here's an example of an npm package published to GitHub Package Registry: https://github.com/o-az/typescript-template/pkgs/npm/typescript-template.

I'm happy to pick up this task and work on it e2e.

**Comment by lochie** [comment]
I agree this would be a wise move. Do you know if there are any GitHub actions that can make this flow automated in some way?

**Reply by o-az** [reply]
I don't know of any existing actions. I've done it writing my own action which is pretty straightforward. I can make a pr for you to see
